plenoptic/tools/conv.py

```diff
@@ -91,3 +95,33 @@ def blur_downsample(x,filtname = 'binom5', step=(2, 2)):
 def upsample_blur(x, step=(2, 2)):
     f = pt.named_filter('binom5')
     return upsample_convolve(x, filt=np.outer(f, f), step=step)
+
+
+def get_same_padding(
+        x: int,
+        kernel_size: int,
+        stride: int,
+        dilation: int
+) -> int:
+    """Helper function to determine integer padding for F.pad() given img and kernel"""
+    pad = (math.ceil(x / stride) - 1) * stride + (kernel_size - 1) * dilation + 1 - x
+    pad = max(pad, 0)
+    return pad
+
+
+def same_padding(
+        x: Tensor,
+        kernel_size: Union[int, Tuple[int, int]],
+        stride: Union[int, Tuple[int, int]] = (1, 1),
+        dilation: Union[int, Tuple[int, int]] = (1, 1),
+        pad_mode: str = "circular",
+) -> Tensor:
+    """Ciruclarly pad an image tensor and output image of same dims."""
```

Comment:
`same_padding` pads the image with values whose entries are dictated by `pad_mode`. E.g. those entries can be filled with zeros, or some other constant, or reflected values of the image, or in this case it will fill the padded entries with values assuming the image is periodic/circular.

Reply:
right, but the docstring makes it sound like it can only do `'circular'`. can we just remove that word? 

And is the reason it's called `same_padding` because the number of pixels it adds is equal to the original size of the image? that is, it doubles the size of `x`?

also, this does assume an image, right? 4d tensor `batch, channel, height, width`. is it worth checking that?

Reply:
Ah didn't see the docstring -- you're right. Fixed. And it assumes at least 2D so it is fully compatible with torch's pad function that doesn't assume 4D.  Added an assertion w/ explanation that the last two dims must be height x width.